lints/lint_dnsname_wildcard_only_in_left_label.go

```diff
@@ -0,0 +1,53 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"strings"
+)
+
+type DNSNameWildcardOnlyInLeftlabel struct {
+	// Internal data here
+}
+
+func (l *DNSNameWildcardOnlyInLeftlabel) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameWildcardOnlyInLeftlabel) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func wildcardNotInLeftLabel(domain string) bool {
+	labels := strings.Split(domain, ".")
+	labels = labels[1:]
+	for _, label := range labels {
+		if strings.Contains(label, "*") {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *DNSNameWildcardOnlyInLeftlabel) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	result := ResultStruct{Result: Pass}
+	if wildcardNotInLeftLabel(c.Subject.CommonName) {
+		result = ResultStruct{Result: Error}
+	}
+	for _, dns := range c.DNSNames {
+		if wildcardNotInLeftLabel(dns) {
```

Comment:
The left label needs to be checked too, it can only be `*`, not `foo*`, or `fo*o`

Reply:
Checked here: https://github.com/zmap/zlint/pull/122